REGRESSION (259976@main): [iOS / macOS] TestWebKitAPI.URLSchemeHandler.DisableCORS is consistently failing

xeenon opened the pull request:
#### 1adec3c9e021542227dd5aa0c187adc9c975fd09
<pre>
REGRESSION (259976@main): [iOS / macOS] TestWebKitAPI.URLSchemeHandler.DisableCORS is consistently failing
<a href="https://webkit.org/b/251935">https://webkit.org/b/251935</a>
rdar://105181176

Reviewed by Sihui Liu.

Remove the dynamic version of _corsDisablingPatterns on WKWebView, since it was was never adopted in Safari
and is not working correctly since SecurityPolicy never removes UserContentURLPatterns from the allowlist.
We are tracking updating this mechanism for extensions in rdar://102912898.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _corsDisablingPatterns]): Deleted.
(-[WKWebView _setCORSDisablingPatterns:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKURLSchemeHandler-1.mm:
Remove test for setting _corsDisablingPatterns on WKWebView.

Canonical link: <a href="https://commits.webkit.org/260048@main">https://commits.webkit.org/260048@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7193d1b79925d83d97a6fa4047014d20c4e1b6da

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106799 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15817 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39641 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115983 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115557 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110702 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17318 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7026 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98988 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112567 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13125 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96152 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40708 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95008 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/27803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9000 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/29158 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9567 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6121 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15169 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48704 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11093 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3749 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (approved):
r=me, we should probably also revert changes in https://commits.webkit.org/224146@main if the functions are not in use any more.

xeenon committed:
REGRESSION (259976@main): [iOS / macOS] TestWebKitAPI.URLSchemeHandler.DisableCORS is consistently failing
https://webkit.org/b/251935
rdar://105181176

Reviewed by Sihui Liu.

Remove the dynamic version of _corsDisablingPatterns on WKWebView, since it was was never adopted in Safari
and is not working correctly since SecurityPolicy never removes UserContentURLPatterns from the allowlist.
We are tracking updating this mechanism for extensions in rdar://102912898.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _corsDisablingPatterns]): Deleted.
(-[WKWebView _setCORSDisablingPatterns:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKURLSchemeHandler-1.mm:
Remove test for setting _corsDisablingPatterns on WKWebView.

Canonical link: https://commits.webkit.org/260048@main
